### DISCUSSION TITLE
Help regarding Flow of internal Sync Flow

### DISCUSSION THREAD

**Discussion by arjunbharara** [Mimetis/Dotmim.Sync#1284] [Q&A]
Which stored procedure is responsible for populating data in the tracking table where the updated_scope_id is set to 0000-0000-0000-0000? This is important as null values are already being handled in triggers.

AND 
which procedure actually selects the changes for Sync.

**Comment by arjunbharara** [comment]
I have a multi-scope setup, and it is generating similar sync_scope_id values for different scopes, which is causing issues. Could you please specify what might be the reason behind the creation of similar sync_scope_id values?

**Reply by Mimetis** [reply]
`0000-0000-0000-0000` is the server scope, always

**Comment by arjunbharara** [comment]
f 0000-0000-0000-0000 is always the server scope, then what is the role of sync_scope_id? How does it function in the overall synchronization process if it ultimately resolves to 0000-0000-0000-0000?

**Reply by Mimetis** [reply]
`sync_scope_id` is uniquely identifying each client with a random GUID generated on the first sync
`sync_scope_id` is always Guid.Empty (0000-......) for server side

**Comment by arjunbharara** [comment]
I have found that the _changes stored procedure is called to fetch changes from both sides. It takes two parameters: timestamp and sync_scope_id. On the client side, the sync_scope_id is sent as null.

Is there any way to intercept or modify this behavior to replace null with a normal ID or a standard GUID?

**Reply by Mimetis** [reply]
If you look at the `GetChanges` method, you have this `WHERE` clause:

``` sql
WHERE (
	[side].[timestamp] > @sync_min_timestamp
	AND ([side].[update_scope_id] <> @sync_scope_id OR [side].[update_scope_id] IS NULL)
)
```

`@sync_min_timestamp` is the inferior limit from which you want the changes.
Usually, this value comes from the `scope_info_client` table

`@sync_scope_id` is the client_id from which one you **dont want** the rows.


**Explanation:**

Each time a database is updating a row  locally, the `update_scope_id` is set to `NULL`: 
 - `NULL` means that this row has been updated locally.

ON the client side, when you try to get changes for a particular client ( for example `ClientId` = `56FGH-DDD....`), you want:

 - **ALL** the rows updated locally (`update_scope_id` == `NULL`)
 - **NOT** the rows where the  `update_scope_id` has been updated to the correct value by the server (when server receive a `NULL` value, it replaces it with the correct client_id), so `@sync_scope_id` != `56FGH-DDD....`

ON the server side, you want the rows updated by the server (NULL) and all the others clients , to be sent to a particular client (`56FGH-DDD....`):
 - **ALL** the rows updated locally (`update_scope_id` == `null`)
 - **ALL** the rows that are different from the client we are targetting `@sync_scope_id` != `56FGH-DDD....`
 - **NOT** the rows updated by the server with a valid client value value ( `@sync_scope_id` != `56FGH-DDD....`)

If you want to see how it's working you can play with the `GetChanges()` method you have on `LocalOrchestrator` and `RemoteOrchestrator`.

That being said, I don't know what you want to achieve by changing the `@sync_scope_id` value, but you may have a better view with this explanation
